### ISSUE TITLE
[flang] No error was reported but setting error on: a (Implicit): ObjectEntity type: REAL(4)

### ISSUE THREAD

**Issue by YuanchengJiang** [llvm/llvm-project#220479]
The following code:

```fortran
character(3), parameter :: arr(3) = [(a(1:i), i=1,3)]
end
```

Resulted in this output:
```
fatal internal error: No error was reported but setting error on: a (Implicit): ObjectEntity type: REAL(4)
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace and instructions to reproduce the bug.
Stack dump:
0.	Program arguments: /cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24 -fc1 -triple x86_64-unknown-linux-gnu -emit-obj -fno-color-diagnostics -mrelocation-model pic -pic-level 2 -pic-is-pie -target-cpu x86-64 -debug-info-kind=standalone -dwarf-version=5 -resource-dir /cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/lib/clang/24 -fintrinsic-modules-path /cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/lib/clang/24/finclude/flang/x86_64-unknown-linux-gnu -mframe-pointer=all -o /tmp/example-b24a58.o -x f95 /app/example.f90
 #0 0x0000000004afded8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4afded8)
 #1 0x0000000004afaf7d SignalHandler(int, siginfo_t*, void*) Signals.cpp:0:0
 #2 0x00007c4fd3e45330 (/lib/x86_64-linux-gnu/libc.so.6+0x45330)
 #3 0x00007c4fd3e9ec0c pthread_kill (/lib/x86_64-linux-gnu/libc.so.6+0x9ec0c)
 #4 0x00007c4fd3e4527e raise (/lib/x86_64-linux-gnu/libc.so.6+0x4527e)
 #5 0x00007c4fd3e288ff abort (/lib/x86_64-linux-gnu/libc.so.6+0x288ff)
 #6 0x0000000008c0ca39 (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x8c0ca39)
 #7 0x000000000562b804 (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x562b804)
 #8 0x000000000562d72b (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x562d72b)
 #9 0x0000000005490ba2 Fortran::evaluate::ExpressionAnalyzer::Analyze(Fortran::parser::ArrayElement const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x5490ba2)
#10 0x000000000548ab4f std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::common::log2visit::Log2VisitHelper<0ul, 3ul, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>>, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>>(std::variant<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>> const&)::'lambda'(auto const&), std::variant<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>> const&>(std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>>(std::variant<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>> const&)::'lambda'(auto const&)&&, unsigned long, std::variant<Fortran::parser::Name, Fortran::common::Indirection<Fortran::parser::StructureComponent, false>, Fortran::common::Indirection<Fortran::parser::ArrayElement, false>, Fortran::common::Indirection<Fortran::parser::CoindexedNamedObject, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x548ab4f)
#11 0x000000000548e851 Fortran::evaluate::ExpressionAnalyzer::Analyze(Fortran::parser::Designator const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x548e851)
#12 0x000000000547b0d8 std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::common::log2visit::Log2VisitHelper<0ul, 7ul, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>>, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&)::'lambda'(auto const&), std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&>(std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&)::'lambda'(auto const&)&&, unsigned long, std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547b0d8)
#13 0x000000000547b26e std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::ExprOrVariable<Fortran::parser::Expr>(Fortran::parser::Expr const&, Fortran::parser::CharBlock) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547b26e)
#14 0x000000000547c161 Fortran::evaluate::ExpressionAnalyzer::Analyze(Fortran::parser::Expr const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547c161)
#15 0x000000000547f42b Fortran::evaluate::ArrayConstructorContext::Add(Fortran::parser::Expr const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547f42b)
#16 0x00000000054841b0 Fortran::evaluate::ArrayConstructorContext::Add(Fortran::parser::AcImpliedDo const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x54841b0)
#17 0x000000000547a93c Fortran::evaluate::ExpressionAnalyzer::Analyze(Fortran::parser::ArrayConstructor const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547a93c)
#18 0x000000000547b0f9 std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::common::log2visit::Log2VisitHelper<0ul, 7ul, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>>, std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&)::'lambda'(auto const&), std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&>(std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&)::'lambda'(auto const&)&&, unsigned long, std::variant<Fortran::common::Indirection<Fortran::parser::CharLiteralConstantSubstring, false>, Fortran::parser::LiteralConstant, Fortran::parser::ConditionalExpr, Fortran::common::Indirection<Fortran::parser::Designator, false>, Fortran::parser::ArrayConstructor, Fortran::parser::StructureConstructor, Fortran::common::Indirection<Fortran::parser::FunctionReference, false>, Fortran::parser::Expr::Parentheses, Fortran::parser::Expr::UnaryPlus, Fortran::parser::Expr::Negate, Fortran::parser::Expr::NOT, Fortran::parser::Expr::PercentLoc, Fortran::parser::Expr::DefinedUnary, Fortran::parser::Expr::Power, Fortran::parser::Expr::Multiply, Fortran::parser::Expr::Divide, Fortran::parser::Expr::Add, Fortran::parser::Expr::Subtract, Fortran::parser::Expr::Concat, Fortran::parser::Expr::LT, Fortran::parser::Expr::LE, Fortran::parser::Expr::EQ, Fortran::parser::Expr::NE, Fortran::parser::Expr::GE, Fortran::parser::Expr::GT, Fortran::parser::Expr::AND, Fortran::parser::Expr::OR, Fortran::parser::Expr::EQV, Fortran::parser::Expr::NEQV, Fortran::parser::Expr::DefinedBinary, Fortran::parser::Expr::ComplexConstructor, Fortran::common::Indirection<Fortran::parser::SubstringInquiry, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547b0f9)
#19 0x000000000547b26e std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::ExprOrVariable<Fortran::parser::Expr>(Fortran::parser::Expr const&, Fortran::parser::CharBlock) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547b26e)
#20 0x000000000547bfdf Fortran::evaluate::ExpressionAnalyzer::IterativelyAnalyzeSubexpressions(Fortran::parser::Expr const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547bfdf)
#21 0x000000000547c225 Fortran::evaluate::ExpressionAnalyzer::Analyze(Fortran::parser::Expr const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x547c225)
#22 0x00000000052207fc std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::evaluate::ExpressionAnalyzer::Analyze<Fortran::common::Indirection<Fortran::parser::Expr, false>>(Fortran::parser::Constant<Fortran::common::Indirection<Fortran::parser::Expr, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x52207fc)
#23 0x0000000005588f78 std::optional<Fortran::evaluate::Expr<Fortran::evaluate::SomeType>> Fortran::semantics::AnalyzeExpr<Fortran::parser::Constant<Fortran::common::Indirection<Fortran::parser::Expr, false>>>(Fortran::semantics::SemanticsContext&, Fortran::parser::Constant<Fortran::common::Indirection<Fortran::parser::Expr, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x5588f78)
#24 0x00000000055890c7 Fortran::semantics::DeclarationVisitor::NonPointerInitialization(Fortran::parser::Name const&, Fortran::parser::Constant<Fortran::common::Indirection<Fortran::parser::Expr, false>> const&) (.part.0) resolve-names.cpp:0:0
#25 0x00000000055a438e Fortran::semantics::DeclarationVisitor::Initialization(Fortran::parser::Name const&, Fortran::parser::Initialization const&, bool) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55a438e)
#26 0x00000000055a8263 Fortran::semantics::DeclarationVisitor::Post(Fortran::parser::EntityDecl const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55a8263)
#27 0x00000000055d481d void Fortran::common::log2visit::Log2VisitHelper<8ul, 15ul, void, void Fortran::parser::detail::ParseTreeVisitorLookupScope::Walk<Fortran::semantics::ResolveNamesVisitor, Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>> const&, Fortran::semantics::ResolveNamesVisitor&)::'lambda'(Fortran::semantics::ResolveNamesVisitor const&), std::variant<Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>> const&>(void Fortran::parser::detail::ParseTreeVisitorLookupScope::Walk<Fortran::semantics::ResolveNamesVisitor, Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>>(std::variant<Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>> const&, Fortran::semantics::ResolveNamesVisitor&)::'lambda'(Fortran::semantics::ResolveNamesVisitor const&)&&, unsigned long, std::variant<Fortran::common::Indirection<Fortran::parser::DerivedTypeDef, false>, Fortran::common::Indirection<Fortran::parser::EnumDef, false>, Fortran::common::Indirection<Fortran::parser::EnumerationTypeDef, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::GenericStmt, false>>, Fortran::common::Indirection<Fortran::parser::InterfaceBlock, false>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::OldParameterStmt, false>>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::ProcedureDeclarationStmt, false>>, Fortran::parser::Statement<Fortran::parser::OtherSpecificationStmt>, Fortran::parser::Statement<Fortran::common::Indirection<Fortran::parser::TypeDeclarationStmt, false>>, Fortran::common::Indirection<Fortran::parser::StructureDef, false>, Fortran::common::Indirection<Fortran::parser::OpenACCDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPDeclarativeConstruct, false>, Fortran::common::Indirection<Fortran::parser::OpenMPMisplacedEndDirective, false>, Fortran::common::Indirection<Fortran::parser::OpenMPInvalidDirective, false>, Fortran::common::Indirection<Fortran::parser::CompilerDirective, false>> const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55d481d)
#28 0x00000000055d8ea1 void Fortran::parser::detail::ParseTreeVisitorLookupScope::Walk<Fortran::parser::DeclarationConstruct, Fortran::semantics::ResolveNamesVisitor>(std::__cxx11::list<Fortran::parser::DeclarationConstruct, std::allocator<Fortran::parser::DeclarationConstruct>> const&, Fortran::semantics::ResolveNamesVisitor&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55d8ea1)
#29 0x00000000055d6978 Fortran::semantics::ResolveNamesVisitor::Pre(Fortran::parser::SpecificationPart const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55d6978)
#30 0x00000000055e2cf3 Fortran::semantics::ResolveNamesVisitor::ResolveSpecificationParts(Fortran::semantics::ProgramTree&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55e2cf3)
#31 0x00000000055f0836 Fortran::semantics::ResolveNamesVisitor::Pre(Fortran::parser::ProgramUnit const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55f0836)
#32 0x00000000055f249c Fortran::semantics::ResolveNames(Fortran::semantics::SemanticsContext&, Fortran::parser::Program const&, Fortran::semantics::Scope&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x55f249c)
#33 0x0000000005685068 Fortran::semantics::Semantics::Perform() (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x5685068)
#34 0x0000000004b612a8 Fortran::frontend::FrontendAction::runSemanticChecks() (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4b612a8)
#35 0x0000000004e5e6e0 Fortran::frontend::CodeGenAction::beginSourceFileAction() (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4e5e6e0)
#36 0x0000000004b66705 Fortran::frontend::FrontendAction::beginSourceFile(Fortran::frontend::CompilerInstance&, Fortran::frontend::FrontendInputFile const&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4b66705)
#37 0x0000000004b4da8f Fortran::frontend::CompilerInstance::executeAction(Fortran::frontend::FrontendAction&) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4b4da8f)
#38 0x0000000004b6de9f Fortran::frontend::executeCompilerInvocation(Fortran::frontend::CompilerInstance*) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x4b6de9f)
#39 0x0000000002cbb6b5 fc1_main(llvm::ArrayRef<char const*>, char const*) (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x2cbb6b5)
#40 0x0000000002c0ac36 main (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x2c0ac36)
#41 0x00007c4fd3e2a1ca (/lib/x86_64-linux-gnu/libc.so.6+0x2a1ca)
#42 0x00007c4fd3e2a28b __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2a28b)
#43 0x0000000002cb9f55 _start (/cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin/flang-24+0x2cb9f55)
flang-24: error: unable to execute command: Aborted (core dumped)
flang-24: error: flang frontend command failed with exit code -2 (use -v to see invocation)
flang version 24.0.0git (https://github.com/llvm/llvm-project.git d770d1ee93075afb93b7b41705daf37144cd3bd6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /cefs/25/25da7b40cec3cf1146176991_clang-llvmflang-trunk-20260901/bin
flang-24: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING CRASH REPRODUCER FILES TO THE BUG REPORT:
flang-24: note: diagnostic msg: /tmp/example-3c9ff3
flang-24: note: diagnostic msg: /tmp/example-3c9ff3.sh
flang-24: note: diagnostic msg: 

********************
flang-trunk - 1985ms
```

Trunk: https://godbolt.org/z/189qYK1EY

Goes back to flang-16: https://godbolt.org/z/4cT644q7n
